### PR TITLE
add azure clean job

### PR DESCRIPTION
**Pull request by mjudeikis**
Fixes https://github.com/openshift/release/issues/970

### CODE REVIEW

File: projects/acs-engine/azure-purge/build.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+kind: BuildConfig
```
Comment (kargakis) [inline projects/acs-engine/azure-purge/build.yaml]: We need to rename acs-engine to azure and have an acs-engine subdirectory. Can be done as a follow-up but open an issue to track please.

Comment (mjudeikis) [reply projects/acs-engine/azure-purge/build.yaml]: https://github.com/openshift/release/issues/988

File: projects/acs-engine/azure-purge/azure-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+### Secret was generated with command:
+### oc create secret generic azure-secret --from-literal=azure_client_id=xxxx --from-literal=azure_client_secret=xxxxx --dry-run -o yaml
```
Comment (kargakis) [inline projects/acs-engine/azure-purge/azure-secret.yaml]: This needs to be part of the Makefile I think

@stevekuznetsov @smarterclayton @dobbymoodge how do we create secrets for other clouds in the CI cluster today? Related: https://trello.com/c/3LlnWrxf/550-ci-automate-secret-creation-in-the-ci-cluster

File: projects/acs-engine/azure-purge/git-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+### to get this secret:
```
Comment (kargakis) [inline projects/acs-engine/azure-purge/git-secret.yaml]: Why is this needed?

File: projects/acs-engine/azure-purge/build.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+apiVersion: v1
+metadata:
+  name: azure-purge
```
Comment (kargakis) [inline projects/acs-engine/azure-purge/build.yaml]: namespace: azure

File: projects/acs-engine/azure-purge/cronjob.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: azure-purge
```
Comment (kargakis) [inline projects/acs-engine/azure-purge/cronjob.yaml]: namespace: azure

File: projects/acs-engine/azure-purge/git-secret.yaml (hunk 3 above)
Comment (mjudeikis) [reply projects/acs-engine/azure-purge/git-secret.yaml]: we build from azure-misc, which is private. so we need to have ssh key to access it (easiest way) or move some of the stuff to public repo

Comment (mjudeikis) [reply projects/acs-engine/azure-purge/git-secret.yaml]: I propose to use ssh key I just added to misc repo and add it as deploy key to only our repo.

File: projects/azure/acs-engine/azure-purge/azure-secret.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+### Secret was generated with command:
```
Comment (kargakis) [inline projects/azure/acs-engine/azure-purge/azure-secret.yaml]: No need for this anymore

Comment (kargakis) [reply projects/azure/acs-engine/azure-purge/azure-secret.yaml]: Sorry, disregard

File: projects/azure/acs-engine/azure-purge/azure-secret.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+### oc create secret generic azure-secret --from-literal=azure_client_id=xxxx --from-literal=azure_client_secret=xxxxx --dry-run -o yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: azure
+  name: azure-secret
```
Comment (kargakis) [inline projects/azure/acs-engine/azure-purge/azure-secret.yaml]: azure-credentials

File: projects/azure/acs-engine/azure-purge/build-artifact.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+kind: BuildConfig
+apiVersion: v1
+metadata:
+  name: azure-purge-artifact
```
Comment (kargakis) [inline projects/azure/acs-engine/azure-purge/build-artifact.yaml]: Our convention re chained builds is to have the build that builds binaries named as $tool-binaries and the build that bind-mounts the binaries in the final image named as $tool. In our case it should be `azure-purge-binaries` and `azure-purge`.

File: projects/azure/acs-engine/azure-purge/is-artifact.yaml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: azure-purge-artifact
```
Comment (kargakis) [inline projects/azure/acs-engine/azure-purge/is-artifact.yaml]: We don't need to separate image streams, elsewhere we use the same but push different tags (`binaries` and `latest`).

**Comment by kargakis** [comment]
Your rebase was semi-successful :) You need to move from `projects/azure/acs-engine/azure-purge` to `projects/azure/azure-purge`.

File: projects/azure/acs-engine/azure-purge/azure-secret.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+### Secret was generated with command:
+### oc create secret generic azure-secret --from-literal=azure_client_id=xxxx --from-literal=azure_client_secret=xxxxx --dry-run -o yaml
```
Comment (kargakis) [inline projects/azure/acs-engine/azure-purge/azure-secret.yaml]: This command needs to move in the Makefile.
